### DISCUSSION TITLE
pg_restore fails due to missing constraint

### DISCUSSION THREAD

**Discussion by dimitrigraf** [netbox-community/netbox#8479] [Q&A] [answered]
Hi Netbox community

I'm trying to restore a dump to a fresh install of netbox but the restore fails with the following message:

```
pg_restore: dropping FK CONSTRAINT virtualization_vminterface virtualization_vminterface_virtual_machine_id_e9f89829_fk
pg_restore: while PROCESSING TOC:
pg_restore: from TOC entry 4553; 2606 24150 FK CONSTRAINT virtualization_vminterface virtualization_vminterface_virtual_machine_id_e9f89829_fk netbox
pg_restore: error: could not execute query: ERROR:  constraint "virtualization_vminterface_virtual_machine_id_e9f89829_fk" of relation "virtualization_vminterface" does not exist
Command was: ALTER TABLE ONLY public.virtualization_vminterface DROP CONSTRAINT virtualization_vminterface_virtual_machine_id_e9f89829_fk;
```

I'm using netbox v3.0.2, installed on a Kubernetes cluster with the netbox helm chart.
The dump was created with `pg_dump -Fc` and the restore with `pg_restore --clean --exit-on-error --verbose`. 
The origin netbox setup should be exactly the same as the one I'm trying to restore the dump to. 
I've also tried to recreate a fresh/empty `netbox` DB in postgres but the result is the same.

Has anybody else stumbled across this error? Or does anyone have an idea how to go forward from here?

Help is very much appreciated. Thanks :)

**Comment by dimitrigraf** [comment, marked as answer]
I solved the issue: I had to change the restore command to `pg_restore --create --clean --exit-on-error --verbose -U postgres -h localhost -d postgres DUMPFILE`

That way `pg_restore` connects to the `postgres` db, from there drops and creates the `netbox` db (as instructed by dump), connects to it and restores the whole dump.

**Reply by jeremystretch** [reply]
Thank you for following up with your solution!

**Comment by Mainstay-Noah-Huppert** [comment]
I had this same error message, but in my case it was because the database I was trying to restore into had run out of disk space.
